src/osd/PrimaryLogPG.cc

```diff
@@ -10382,32 +10382,22 @@ void PrimaryLogPG::issue_repop(RepGather *repop, OpContext *ctx)
     }
   }
 
-  for (set<pg_shard_t>::const_iterator i = acting_recovery_backfill.begin();
-       i != acting_recovery_backfill.end();
-       ++i) {
-    pg_shard_t peer(*i);
-    if (peer == pg_whoami) continue;
-    if (async_recovery_targets.count(peer) && peer_missing[peer].is_missing(soid)) {
-      for (auto &&entry: ctx->log) {
-	missing_loc.add_missing(soid, ctx->at_version, eversion_t(), entry.is_delete());
-      }
-    }
-  }
-
-  dout(30) << __func__ << " missing_loc before: " << missing_loc.get_locations(soid) << dendl;
-
   if (requires_missing_loc) {
```

Comment:
@xiexingguo This looks more like using `requires_missing_loc` to update `missing_loc`, which looks fine to me. Not sure, I understand the commit description "There could be clones and hence we should update missing_loc
based on log entry itself".

Reply:
No, that is not the key point.